pkg/converter/converter_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package converter
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/istio-ecosystem/sail-operator/pkg/test/project"
+	"github.com/istio-ecosystem/sail-operator/pkg/test/util/supportedversion"
+	"github.com/istio-ecosystem/sail-operator/tests/e2e/util/shell"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	controlPlaneNamespace = "istio-system"
+	istioVersion          = supportedversion.Default
+	istioFile             = filepath.Join(project.RootDir, "tools", "istioConfig.yaml")
+	sailFile              = filepath.Join(project.RootDir, "tools", "istioConfig-sail.yaml")
+)
+
+func TestConversion(t *testing.T) {
+	testcases := []struct {
+		name           string
+		input          string
+		expectedOutput string
+		converterArgs  string
+	}{
+		{
+			name: "simple",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`,
+			expectedOutput: fmt.Sprintf(`apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s`, istioVersion, controlPlaneNamespace),
+			converterArgs: fmt.Sprintf("%s %s -n %s -v %s", istioFile, sailFile, controlPlaneNamespace, istioVersion),
```

Comment:
Is there an actual benefit in using variables instead of using the simpler approach of just including the values in the YAML and args? The following takes much less effort to read:
```suggestion
			expectedOutput: `apiVersion: sailoperator.io/v1
kind: Istio
metadata:
  name: default
spec:
  version: v1.24.0
  namespace: my-namespace`,
			converterArgs: "/tmp/input.yaml /tmp/output.yaml -n my-namespace -v v1.24.0",
```